[ADD] HR - First public version of new module with the data mode adaptations to the Portuguese law

andreparames opened the pull request:
This module adds many new fields and mechanisms to the different HR models (Employee, Contract, Payslip, Salary Rules and others), adapting Odoo to the Portuguese legislation as with regard to HR.

Commit:
Added first public version of tko_hr_pt

Commit:
Added first public version of tko_hr_pt

andreparames commented:
Sorry for the clabot above, we hadn't linked the git author to the Github user yet, so it didn't detected that we had signed the CLA.

dreispt commented on tko_hr_pt/__init__.py:
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
```
Module names should not identify the company name.
I suggest `l10n_pt_hr`, following what is done in other localizations.

dreispt commented on tko_hr_pt/__init__.py:
```diff
@@ -0,0 +1,28 @@
+#
+########################################################################################
+
+import base
```
`from . import ...`

dreispt commented on tko_hr_pt/__init__.py:
```diff
@@ -0,0 +1,28 @@
+########################################################################################
+#
+# Copyright (C) ThinkOpen Solutions (<http://thinkopen.solutions>). All Rights Reserved
+# Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>). All Rights Reserved
```
IMO you should remove this line: Tiny is not an author of this module.

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'version': '1.064',
+    'category': 'Human Resources',
+    "sequence": 38,
+    'complexity': "normal",
```
Complexity deprecated

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'category': 'Human Resources',
+    "sequence": 38,
+    'complexity': "normal",
+    'description': """
```
Can be removed and replaced by a README.rst or .md

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'depends': [
+                'base','hr_payroll',
+    ],
+    'init_xml': [
```
remove init_xml

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    ],
+    'init_xml': [
+    ],
+    'update_xml': [
```
deprecated, use `data`

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    ],
+    'test': [
+    ],
+    'demo_xml': [
```
demo_xml deprecated

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'demo_xml': [
+    ],
+    'installable': True,
+    'active': False,
```
active deprecated

dreispt commented on tko_hr_pt/base.py:
```diff
@@ -0,0 +1,37 @@
+    _columns = {
+                'identification_name': fields.char('Identification Name', size=64),
+                'niss': fields.char('SS Identification', size=64),
+                'nif': fields.char('Tax Identification', size=64),
```
A company is also a partner, so it already has a VAT number, so `nif` is redundant.
To avoid break depending functionality, I suggest making it a related field.

dreispt commented on tko_hr_pt/__init__.py:
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+# coding: utf-8
```
`# -*- coding: utf-8 -*-`

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'complexity': "normal",
+    'description': """
+    """,
+    'author':"ThinkOpen Solutions PT",
```
Add `,Odoo Community Association (OCA)` (to allow searching for OCA modules in the apps site)

dreispt commented on tko_hr_pt/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'installable': True,
+    'active': False,
+    'application': False,
+}
```
Add the license key stating LGPL or it will show as AGPL

andreparames replied:
Apparently "LGPL" isn't a valid key for the license field. Should we use "Other OSI approved licence"?

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+                    continue
+                index = next(index for (index, d) in enumerate(self._history_vals_log) if d.has_key(value))
+                object = self._history_vals_log[index][value]
+                change = '%s - Referencia contrato %s' % (vals[value], line.name)
```
Make string translatable

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+class hr_employee(osv.osv):
+    '''
+    Employee
+    '''
```
I think triple single quotes don't pass pep8 checks...

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+                                         ('E - Rendimentos de capitais', 'E - Rendimentos de capitais'),
+                                         ('F - Rendimentos prediais', 'F - Rendimentos prediais'),
+                                         ('G - Incrementos patrimoniais', 'G - Incrementos patrimoniais'),
+                                         ('H - Pensões', 'H - Pensões'),
```
It would be wiser for the selection  identifiers to be just the code ("A", "B", ...) ...

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+            raise osv.except_osv(_('Error !'), _('Cannot select %s with marital: %s') %(self._ownership[ownership],self._marital[marital]))
+        return res
+
+hr_employee()
```
deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+    _columns = {
+                'total': fields.function(_calculate_total, method=True, type='float', string='Total', digits_compute=dp.get_precision('Payroll'),store=True ),
+                'company_rate' : fields.float(string='Company Rate',digits_compute=dp.get_precision('Payroll')),
+###
```
remove line

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+#         if category.parent_id:
+#             localdict = self._sum_salary_rule_category(localdict, category.parent_id, amount)
+#         localdict['categories'].dict[category.code] = category.code in localdict['categories'].dict and localdict['categories'].dict[category.code] + amount or amount
+#         return localdict
```
remove this commented code

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+              'user_id':fields.many2one('res.users', 'User', required=True),
+              }
+    
+hr_employee_history()
```
deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt.py:
```diff
@@ -0,0 +1,950 @@
+                end_date =  date(int(compute_date_year) + 1, 1, 1)
+            payslip_ids = slip_pool.search(cr, uid, [
+                                                     ('date_from','>=',str(start_date))
+                                                     ,('date_from','<',str(end_date))
```
str(date) is dangerous. Proper date to string should be used.

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+        <record id="tko_view_employee_form_inherit" model="ir.ui.view">
+            <field name="name">tko.view.employee.history.form.inherit</field>
+            <field name="model">hr.employee</field>
+            <field name="type">form</field>
```
deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+                </field>
+                <page string="HR Settings" position="after">
+                    <page string="History" groups="base.group_hr_user">
+                        <field name="history_lines" colspan="4" nolabel="1">
```
colspan deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+        <record id="hr_employee_view_inherit_form" model="ir.ui.view">
+            <field name="name">hr.employee.view.inherit.form</field>
+            <field name="model">hr.employee</field>
+            <field name="type">form</field>
```
type deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+                    </field>
+                     <field name ="marital" position="replace">
+                    	<field name = "marital" on_change="onchange_marital(marital)"/>
+                    </field>-->
```
remove dead code

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+<!--                     <field name="sequence"/> -->
+<!--                 </field> -->
+<!--             </field> -->
+<!--         </record> -->
```
remove dead code

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+                    <button string="Set to Draft" type='workflow' name="draft" states="cancel"/>
+                 </button>
+                 <field name="input_line_ids" position="after">
+                    <separator string="Ad-Hoc Rules" colspan="4" groups="tko_hr_pt.group_adhoc_rules"/>
```
colspan deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+                    <field name="date_from"/>
+                    <filter icon="terp-document-new" string="Draft" domain="[('state','=','draft')]" help="Draft Slip"/>
+                    <filter icon="terp-camera_test" string="Done" domain="[('state','=','done')]" help="Done Slip"/>
+                    <filter icon="terp-camera_test" string="Cancel" domain="[('state','=','cancel')]" help="Cancel Slip"/>
```
afaik icons on search views are deprecated

dreispt commented on tko_hr_pt/hr_payroll_pt_view.xml:
```diff
@@ -0,0 +1,565 @@
+            <field name="model">hr.social.security.pt</field>
+            <field name="arch" type="xml">
+                <form string="Social Security " version="7.0">
+                    <sheet string=" ">
```
string=" "?

dreispt commented:
Welcome to OCA and thank you for publishing this!

Please note that it's normal to have some "pains" to adjust to the sctrict OCA code quality rules.
But in the end it will pay of, and will get easier as we get used to them.

TravisCI is [complaining](https://travis-ci.org/OCA/l10n-portugal/jobs/58312091) about PEP8.
Some work needed there. You might want to consider the autopep8 tool, and also this [helper script](https://github.com/OCA/maintainer-quality-tools/tree/master/git).

max3903 commented:
CLA problem should be fixed.

andreparames commented:
Hi @dreispt, thanks a lot for the in-depth review, we'll get on to fixing these issues and will resubmit once they are fixed. 

Thanks to you too, @max3903.

gurneyalex committed:
Update travis configuration

Follow OCA/maintainer-quality-tools#187 and use addons/apt/package directive to
install our build dependencies. This allows using container based Travis builds
(which should be faster).

andreparames committed:
Changes to comply with PEP8 and removing obsolete attributes

andreparames committed:
Changes to comply with PEP8 and removing obsolete attributes

andreparames committed:
Set the fields required in the view instead of in the model

andreparames committed:
Set the fields required in the view instead of in the model

andreparames committed:
Merge branch '8.0' of github.com:ThinkOpen-Solutions/l10n-portugal into 8.0

andreparames committed:
Updated license in __openerp__.py to match one of the allowed values in ir.module.module.license

andreparames commented:
Sorry, is the runbot working correctly? I seem to be unable to open its logs, and it says the runtime is 0s, which seems low :)

dreispt commented:
At least until recently, OCA Runbot was not perfect. So don't worry too much about it for now.

dreispt commented:
@andreparames I'm helping to push this forward. is this the latest version or do you have any later work you can share?

pedrobaeza commented:
Closing this as no further notices. It's a pity to not have more from Portuguese community, but we can't either accept modules that don't pass CIs.

andreparames commented:
ping @hmrodrigues
